### PR TITLE
Use stored config in all applicable commands if not specified explicitly

### PR DESCRIPTION
**Pull request by nathanaelhoun**
## Changes

- [x] Use stored config in all aplicable commands if project/org is not specified explicitly
- [x] Move some logging errors to use `p.API.Log*` instead of `&model.AppError` when was seeing them. But I can't garantee I caught them all. Related to #23.
- [x] Update docs (README.md and HOW_TO.md)

### CODE REVIEW

File: server/plugin/command.go [hunk 1]
```diff
@@ -127,27 +129,73 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 		return p.sendEphemeralResponse(args, notConnectedText), nil
 	}
 
+	config, err := p.Store.GetConfig(args.UserId)
+	if err != nil {
+		p.API.LogError("Could not get user config", "error", err)
+	}
+
+	var splitWithoutProject []string
+	if config != nil {
```
Comment (darkLord19) [inline server/plugin/command.go]: I think preference should be given to arguments passed by user via circleci command execution and it it doesn't exist then fallback to default. In this case it, if i am understanding correctly, even if user gives an input, we will always use stored project slug which is not very good experience.

Comment (nathanaelhoun) [reply server/plugin/command.go]: Yes, it a bad behaviour I have removed since and have just pushed

**Comment by nathanaelhoun** [comment]
new-command

**Comment by darkLord19** [comment]
We should do something like this,
- have custom type ProjectSlug{vsc, org, repo}
- Config type will have instance of this ProjectSlug: type Config {projectSlug} //This way we could add any other configs also for default use for future usecases
- for commands which needs project slug or orgslug, do input validation and see if any args are passed for project or org slug value. If not then try to get from stored config. If stored config found then use it otherwise return with error.

@nathanaelhoun let me know your thoughts

**Comment by nathanaelhoun** [comment]
> We should do something like this,
> 
>     * have custom type ProjectSlug{vsc, org, repo}
> 
>     * Config type will have instance of this ProjectSlug: type Config {projectSlug} //This way we could add any other configs also for default use for future usecases
> 
>     * for commands which needs project slug or orgslug, do input validation and see if any args are passed for project or org slug value. If not then try to get from stored config. If stored config found then use it otherwise return with error.
> 
> 
> @nathanaelhoun let me know your thoughts

It's a good idea, but I don't think it's worth the effort. Only a few command require to use the orgslug. It's not bad for user experience if he have to specitify the orgslug sometimes IMO

**Comment by darkLord19** [comment]
> > We should do something like this,
> > ```
> > * have custom type ProjectSlug{vsc, org, repo}
> > 
> > * Config type will have instance of this ProjectSlug: type Config {projectSlug} //This way we could add any other configs also for default use for future usecases
> > 
> > * for commands which needs project slug or orgslug, do input validation and see if any args are passed for project or org slug value. If not then try to get from stored config. If stored config found then use it otherwise return with error.
> > ```
> > 
> > 
> > @nathanaelhoun let me know your thoughts
> 
> It's a good idea, but I don't think it's worth the effort. Only a few command require to use the orgslug. It's not bad for user experience if he have to specitify the orgslug sometimes IMO

agreed.

**Comment by darkLord19** [comment]
Let's add debug/info logs as well in case of successful termination of command

**Review by darkLord19 — APPROVED** [review]
Haven't tested changes myself but logically looks fine. Just verify all commands are working as expected before merging(except workflow get command for which I have raised a new PR).

**Comment by darkLord19** [comment]
> Let's add debug/info logs as well in case of successful termination of command

we can add this later via other PR. This PR had become big as it is.

**Pull request merged**